idioms/ctor.md

```diff
@@ -3,34 +3,75 @@
 ## Description
 
 Rust does not have constructors as a language construct. Instead, the
-convention is to use a static `new` method to create an object.
-
-## Example
-
-```rust,ignore
-// A Rust vector, see liballoc/vec.rs
-pub struct Vec<T> {
-    buf: RawVec<T>,
-    len: usize,
-}
-
-impl<T> Vec<T> {
-    // Constructs a new, empty `Vec<T>`.
-    // Note this is a static method - no self.
-    // This constructor doesn't take any arguments, but some might in order to
-    // properly initialise an object
-    pub fn new() -> Vec<T> {
-        // Create a new Vec with fields properly initialised.
-        Vec {
-            // Note that here we are calling RawVec's constructor.
-            buf: RawVec::new(),
-            len: 0,
-        }
+convention is to use an associated `new` function to create an object:
+
+```rust
+pub struct Second {
+    value: u64
+}
+
+impl Second {
+    pub fn new(value: u64) -> Self {
+        Self { value }
+    }
+}
+
+fn main() {
+    let s = Second::new(42);
+}
+```
+
+## Default Constructors
+
+Rust supports default constructors with the [`Default`][std-default] trait:
+
+```rust
+pub struct Second {
+    value: u64
+}
+
+impl Default for Second {
+    fn default() -> Self {
+        Self { value: 0 }
     }
 }
+
+fn main() {
+    let default = Second::default();
```

Comment:
Are the rustdoc examples tested with mdbook?

Reply:
I have pushed a commit with the `fn main` replaced with rustdoc comments.

Reply:
I'm not sure how I feel in general about this. The clunky rustdoc comments distract from the code a lot. The `fn main` didn't in that magnitude.

There is already a **Note:** and a **Hint:**. I guess it might be best te revert to `fn main` for testable examples, maybe even hide them to keep the code examples very tight, and to add an additional note, explaining that the used defaults should be documented with a rustdoc comment.

Reply:
Think it's fine like this, tbh.

Reply:
I just think we should add a note saying that public fields should be documented so that others know what is the default value. Something like this.

```rust
/// user age, default 0
age: i32,
```

Reply:
there are no public fields here :(